pkg/mcp/types.go

```diff
@@ -296,3 +296,25 @@ func ProjectServerToConfig(tokenService *ephemeral.TokenService, projectMCPServe
 		Runtime:      types.RuntimeRemote,
 	}, nil
 }
+
+func CompositeServerToConfig(tokenService *ephemeral.TokenService, mcpServerName, baseURL, userID, scope string) (ServerConfig, error) {
+	token, err := tokenService.NewToken(ephemeral.TokenContext{
+		UserID:     userID,
+		UserGroups: []string{types.GroupBasic},
+	})
+	if err != nil {
+		return ServerConfig{}, fmt.Errorf("failed to create token: %w", err)
+	}
+
+	finalScope := scope
+	if !strings.HasPrefix(scope, mcpServerName+"-") {
+		finalScope = fmt.Sprintf("%s-%s", mcpServerName, scope)
+	}
```

Comment:
I think the scope should be what the scope is at this point. We don't need to modify it.